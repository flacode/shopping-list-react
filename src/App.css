body,
html {
    height: 100%;
    font-size: 16px;
}

.card-container.card {
  max-width: 450px;
  padding: 40px 40px;
}

.card {
  background-color: #F7F7F7;
  /* just in case there no content*/
  padding: 20px 25px 30px;
  margin: 0 auto 25px;
  margin-top: 50px;
  /* shadows and rounded borders */
  -moz-border-radius: 2px;
  -webkit-border-radius: 2px;
  border-radius: 2px;
  -moz-box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);
  -webkit-box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);
}

.thick-heading {
  clear: both;
  color:rgb(34, 180, 158);
  display: block;
  text-align: center;
  font-weight: 700;
  text-shadow: 0 0 0 rgba(0, 0, 0, 0);
  line-height: 30px;
  border-width: 0;
  font-size: 30px;
  visibility: visible;
  opacity: 1;
  margin-top: 5px;
}

.name-card {
  font-size: 25px;
  font-weight: bold;
  text-align: left;
  min-height: 1em;
  color:rgb(34, 180, 158);
}

.img-card {
  width: 100px;
  height: 100px;
  margin: 0 auto 5px;
  display: block;
  -moz-border-radius: 30%;
  -webkit-border-radius: 30%;
  border-radius: 30%;
}

.form-signin .form-control:focus {
  border-color: rgb(104, 145, 162);
  outline: 0;
  -webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, .075), 0 0 8px rgb(104, 145, 162);
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, .075), 0 0 8px rgb(104, 145, 162);
}

.btn.btn-auth {
  background-color: rgb(34, 180, 158);
  padding: 0px;
  font-weight: 550;
  font-size: 20px;
  height: 36px;
  -moz-border-radius: 3px;
  -webkit-border-radius: 3px;
  border-radius: 3px;
  border: none;
  -o-transition: all 0.218s;
  -moz-transition: all 0.218s;
  -webkit-transition: all 0.218s;
  transition: all 0.218s;
  width: 110px;
}

.btn.btn-auth:hover,
.btn.btn-auth:active,
.btn.btn-auth:focus {
  background-color: rgba(18, 243, 209, 0.5);
}

.btn.btn-signin {
  width: 110px;
}

.login {
  margin-top: 5px;
  color: rgb(104, 145, 162);
}

.auth-link {
  color: rgb(34, 180, 158)
}

.auth-reset {
  font-size: 13px;
}

.btn-center {
  margin:auto;
  display:block;
  text-align:center;
}

.list-page {
  margin-top: 15px;
}

.panel-body {
  height: 100%;
  background-color: #F7F7F7;
  min-height: 80vh;
}

.page-heading {
    font-size: 25px;
    font-weight: bold;
    text-align: left;
    min-height: 1em;
    vertical-align: middle; 
}

.panel-heading.site-background,
.panel-footer.site-background,
.modal-heading {
  background-color: rgb(34, 180, 158);
  color: #F7F7F7
}

.list-group {
  margin-top: 15px;
  margin-bottom: 15px;
  width: 70%; margin-right: auto; margin-left: auto; height: auto;
}

.list-name {
  color: rgb(34, 180, 158);
}

.btn-list {
  width: 250px !important;
  margin:auto;
  display:block;
}

.list-details {
  text-align: left;
}

.action-btn *{
  display:inline;
  margin: auto;
}

.float-left {
  float: left
}

.float-right {
  float: right;
}

.clear-float {
  clear: both;
}

.icon-btn {
  background-color: rgb(34, 180, 158);
  color: #F7F7F7;
  border-color: rgb(34, 180, 158);
}

.link-btn:visited,
.link-btn:hover,
.link-btn:active {
  text-decoration: none;
}

.label-text {
  color: rgb(103, 132, 143);
}

.form-error {
  color: red;
  font-size: 15px;
}

.not-bought {
  color: red;
}

.bought {
  color: rgb(34, 180, 158);
}
.rc-pagination-item {
  color: rgb(34, 180, 158);
  border-color: rgb(34, 180, 158);
}
.rc-pagination-item a {
  text-decoration: none;
  color: rgb(103, 132, 143);
}
.rc-pagination-item:hover {
  border-color: rgb(34, 180, 158);
}
.rc-pagination-item:hover a {
  color: rgb(34, 180, 158);
}
.rc-pagination-item-active {
  background-color: rgb(34, 180, 158);
  border-color: rgb(34, 180, 158);
  color: #fff;
}

.rc-pagination-jump-prev:hover:after,
.rc-pagination-jump-next:hover:after {
  color: rgb(34, 180, 158);
}

.rc-pagination-options-quick-jumper input:hover {
  border-color: rgb(34, 180, 158);
}

.rc-pagination-options-quick-jumper button:hover,
.rc-pagination-options-quick-jumper button:active,
.rc-pagination-options-quick-jumper button:focus {
  color: rgb(34, 180, 158);
  background-color: #fff;
  border-color: rgb(34, 180, 158);
}

.rc-pagination-simple .rc-pagination-simple-pager input:hover {
  border-color: rgb(34, 180, 158);
}

.rc-pagination-simple .rc-pagination-simple-pager button:hover,
.rc-pagination-simple .rc-pagination-simple-pager button:active,
.rc-pagination-simple .rc-pagination-simple-pager button:focus {
  color: rgb(34, 180, 158);
  background-color: #fff;
  border-color: rgb(34, 180, 158);
}
