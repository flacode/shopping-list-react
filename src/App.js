import React, { Component } from 'react';
import ShoppingListTable from './shoppingList';
import ItemTable from './shoppingListItems';
import RegistrationForm from './registrationForm';
import LoginForm from './loginForm';
//import Login from './Login';
//import NewShoppingList from './createShoppingList';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <RegistrationForm />
        <LoginForm />
        <h1>Sample Shopping Lists</h1>
        <ShoppingListTable />
        <h1> Sample shopping list items</h1>
        <ItemTable />
        {/*<br/>
        <br/>
        <Login />
        <br/>
        <br/>
        <NewShoppingList items={items} />
        <ShoppingListTable shoppingLists={shoppingLists} />
        <br/>
        */}
      </div>
    );
  }
}

export default App;
