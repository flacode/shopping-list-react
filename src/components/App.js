import React, { Component } from 'react';
import ShoppingListDashboard from './shoppingList';
import ItemDashBoard from './shoppingListItems';
import {Grid} from 'react-bootstrap';
//import Login from './Login';
//import NewShoppingList from './createShoppingList';
import '../App.css';

class App extends Component {
  render() {
    return (
      <Grid>
        <ShoppingListDashboard />
        <h1> Sample shopping list items</h1>
        <ItemDashBoard />
        {/*<br/>
        <br/>
        <Login />
        <br/>
        <br/>
        <NewShoppingList items={items} />
        <ShoppingListTable shoppingLists={shoppingLists} />
        <br/>
        */}
      </Grid>
    );
  }
}

export default App;
