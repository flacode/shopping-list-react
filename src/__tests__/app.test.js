import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import '../setupTests';
import App from '../components/App';

describe('<App />', () => {
  it('renders without crashing', () => {
    shallow(<App />);
  });
  it('has a valid snapshot', () => {
    const component = renderer.create(
      <App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
