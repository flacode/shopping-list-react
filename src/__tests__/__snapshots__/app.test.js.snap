// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> has a valid snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="card-container card"
    >
      <h5
        className="thick-heading card-title"
      >
        SHOPPING LIST
      </h5>
      <img
        alt="Card image cap"
        className="img-card card-img"
        src="shoppinglist.png"
      />
      <div
        className="card-body"
      >
        <h3
          className="name-card"
        >
          Sign up
        </h3>
        
        <form
          className="form-signin"
        >
          <div
            className="form-group"
          >
            <div>
              <input
                className="form-control"
                name="username"
                onChange={[Function]}
                placeholder="Username"
                type=""
                value=""
              />
              <div
                className="invalid-feedback"
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <div>
              <input
                className="form-control"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="email"
                value=""
              />
              <div
                className="invalid-feedback"
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <div>
              <input
                className="form-control"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
              <div
                className="invalid-feedback"
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <div>
              <input
                className="form-control"
                name="confirmPassword"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
                value=""
              />
              <div
                className="invalid-feedback"
              />
            </div>
          </div>
          <div
            className="btn-center"
          >
            <button
              className="btn-auth btn btn-secondary disabled"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              Submit
            </button>
             
            <button
              className="btn-auth btn btn-secondary"
              onClick={[Function]}
              type="button"
            >
              Reset
            </button>
          </div>
        </form>
        <br />
        <p
          className="login"
        >
          Already have an account, please 
          <a
            className="auth-link"
            href="/login"
            onClick={[Function]}
          >
            login
          </a>
          .
        </p>
      </div>
    </div>
  </div>
</div>
`;
